source/internal/_cloneRegExp.js

```diff
@@ -1,7 +1,8 @@
 export default function _cloneRegExp(pattern) {
-  return new RegExp(pattern.source, (pattern.global     ? 'g' : '') +
-                                    (pattern.ignoreCase ? 'i' : '') +
-                                    (pattern.multiline  ? 'm' : '') +
-                                    (pattern.sticky     ? 'y' : '') +
-                                    (pattern.unicode    ? 'u' : ''));
+  return new RegExp(pattern.source, (pattern.flags ?  pattern.flags : (pattern.global     ? 'g' : '') +
+                                                                      (pattern.ignoreCase ? 'i' : '') +
+                                                                      (pattern.multiline  ? 'm' : '') +
+                                                                      (pattern.sticky     ? 'y' : '') +
+                                                                      (pattern.unicode    ? 'u' : '') +
+                                                                      (pattern.dotAll     ? 's' : '')));
 }
```

Comment:
This looks great.  But could you try the trick mentioned by @adispring to add some test cases?

Reply:
i try it, and test is passed.